File: src/service/WalletService.java
package service;

import model.Wallet;

import java.util.Scanner;

public class WalletService {
    public static void printWallet (Wallet wallet) {
        System.out.println("Wallet: " + "\n  owner: " + wallet.getOwner() + "\n  sold: " + wallet.getSold());
    }

    public static void updateWallet (Wallet wallet, Scanner scanner) {
        System.out.println("Your name: ");
        wallet.setOwner(scanner.nextLine());
        System.out.println("Sold: ");
        wallet.setSold(scanner.nextDouble());
        System.out.println("Update successfully");
    }

    public static void withdrawal (Wallet wallet, Scanner scanner) {
        System.out.println("Amount to withdraw: ");
        double amount = scanner.nextInt();
        if (amount > wallet.getSold()) {
            System.out.println("Insufficient balance");
        }
        else {
            wallet.setSold(wallet.getSold() - amount);
            System.out.println("withdrawal successfully");
        }
    }
}

File: src/service/wallet.service.ts
import { Wallet } from "../model/wallet.model";
import { myPrompt } from "../util/my.prompt.util";

export const printWallet = (wallet: Wallet): void => {
  console.log(`Wallet: \n - sold: ${wallet.getSold()}`);
}

export const deposit = async (wallet: Wallet) => {
  const input = await myPrompt("Money to deposit: ");
  if (!isNaN(Number(input))) {
    wallet.setSold(wallet.getSold() + Number(input));
    return Promise.resolve("Deposit successfully");
  } else {
    return Promise.reject(new Error("Invalid Input - retry"));
  }
}

export const withdrawal = async (wallet: Wallet) => {
  const input = await myPrompt("Amount to withdraw: ");
  if (!isNaN(Number(input))) {
    const amount = Number(input);
    if (amount > wallet.getSold()) {
      return Promise.reject(new Error("Insufficient balance"));
    } else {
      wallet.setSold(wallet.getSold() - amount);
      return Promise.resolve("Withdrawal successfully");
    }
  } else {
    return Promise.reject(new Error("Invalid Input - retry"));
  }
}